File: algorithms/python/1/main.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        for i, n in enumerate(nums):
            if target - n in nums[i + 1:]:
                return [i, nums[i + 1:].index(target - n) + (i + 1)]

    def twoSum2(self, nums: list[int], target: int) -> list[int]:
        nums_map = collections.defaultdict(int)
        for i, num in enumerate(nums):
            nums_map[num] = i

        for i, num in enumerate(nums):
            if target - num in nums_map and i != nums_map[target - num]:
                return i, nums_map[target - num]

File: algorithms/java/1/main.java
import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashSet<Integer> numSet = new HashSet<Integer>();
        HashMap<Integer, Integer> indexMap = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i ++){
            indexMap.put(nums[i], i);
            numSet.add(nums[i]);
        }
        for(int i = 0; i < nums.length; i++){
            if (numSet.contains(target - nums[i]) && i != indexMap.get(target - nums[i])){
                return new int[] {i, indexMap.get(target - nums[i])};
            }
        }
        return new int[] {1, 2};
    }
}